crates/core/component/ibc/src/component/grpc_query.rs

```diff
@@ -0,0 +1,637 @@
+
+        let (conn, proof) = snapshot
+            .get_with_proof_to_apphash(
+                state_key::connections::by_connection_id(connection_id)
+                    .as_bytes()
+                    .to_vec(),
+            )
+            .await
+            .map(|res| {
+                let conn = res
+                    .0
+                    .map(|conn_bytes| ConnectionEnd::decode(conn_bytes.as_ref()))
+                    .transpose();
+
+                (conn, res.1)
+            })
+            .map_err(|e| tonic::Status::aborted(format!("couldn't get connection: {e}")))?;
+
+        let conn =
+            conn.map_err(|e| tonic::Status::aborted(format!("couldn't decode connection: {e}")))?;
+
+        let height = Height {
+            revision_number: 0,
+            revision_height: snapshot.version().into(),
+        };
+
+        let res = QueryConnectionResponse {
+            connection: conn,
+            proof: proof.encode_to_vec(),
+            proof_height: Some(height),
+        };
+
+        Ok(tonic::Response::new(res))
+    }
+    /// Connections queries all the IBC connections of a chain.
+    async fn connections(
+        &self,
+        _request: tonic::Request<QueryConnectionsRequest>,
+    ) -> std::result::Result<tonic::Response<QueryConnectionsResponse>, tonic::Status> {
+        let snapshot = self.0.latest_snapshot();
+        let height = snapshot.version();
+
+        let connection_counter = snapshot
+            .get_connection_counter()
+            .await
+            .map_err(|e| tonic::Status::aborted(format!("couldn't get connection counter: {e}")))?;
+
+        let mut connections = vec![];
+        for conn_idx in 0..connection_counter.0 {
+            let conn_id = ConnectionId(format!("connection-{}", conn_idx));
+            let connection = snapshot
+                .get_connection(&conn_id)
+                .await
+                .map_err(|e| {
+                    tonic::Status::aborted(format!("couldn't get connection {conn_id}: {e}"))
+                })?
+                .unwrap();
+            let id_conn = IdentifiedConnectionEnd {
+                connection_id: conn_id,
+                connection_end: connection,
+            };
+            connections.push(id_conn.into());
+        }
+
+        let height = Height {
+            revision_number: 0,
+            revision_height: height.into(),
+        };
+
+        let res = QueryConnectionsResponse {
+            connections,
+            pagination: None,
+            height: Some(height),
+        };
+
+        Ok(tonic::Response::new(res))
+    }
+    /// ClientConnections queries the connection paths associated with a client
+    /// state.
+    async fn client_connections(
+        &self,
+        _request: tonic::Request<QueryClientConnectionsRequest>,
+    ) -> std::result::Result<tonic::Response<QueryClientConnectionsResponse>, tonic::Status> {
+        todo!()
```

Comment:
This will make the node panic if we get one of these requests, can we search/replace these `todo!()` impls with a snippet that returns an `Ok` with [`Status::unimplemented`](https://docs.rs/tonic/latest/tonic/struct.Status.html#method.unimplemented) ?